profiling/src/profile/internal/mapping.rs

```diff
@@ -0,0 +1,69 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present Datadog, Inc.
+
+use super::{Id, Item, PprofItem};
+use crate::profile::{pprof, StringId};
+use std::num::NonZeroU32;
+
+#[derive(Eq, PartialEq, Hash)]
+pub struct Mapping {
+    /// Address at which the binary (or DLL) is loaded into memory.
+    pub memory_start: u64,
+    /// The limit of the address range occupied by this mapping.
+    pub memory_limit: u64,
```

Comment:
As it stands now, we're stuck with this; but if we change from a `[start, end)` formulation to a `[start, start + size)` formulation, we could realistically squeeze the `size` parameter into a u32.  These mappings are actually ELF sections (otherwise including both start + file offset is kinda weird), and they probably need to be _executable_ unless we're providing location information for non-code resources (why lol), soooo I personally think u32 would be reasonable there.

Obviously this would be inappropriate for this PR, just making a comment.

Reply:
What about needing to be _executable_ makes it reasonable to expect it to fit in a u32?